Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserTableModel.cpp

```diff
@@ -24,13 +24,24 @@ namespace AzToolsFramework
             AZ_Assert(
                 m_filterModel,
                 "Error in AssetBrowserTableModel initialization, class expects source model to be an AssetBrowserFilterModel.");
+
             QSortFilterProxyModel::setSourceModel(sourceModel);
 
-            connect(m_filterModel, &QAbstractItemModel::rowsInserted, this, &AssetBrowserTableModel::UpdateTableModelMaps);
-            connect(m_filterModel, &QAbstractItemModel::rowsRemoved, this, &AssetBrowserTableModel::UpdateTableModelMaps);
-            connect(m_filterModel, &QAbstractItemModel::layoutChanged, this, &AssetBrowserTableModel::UpdateTableModelMaps);
-            connect(m_filterModel, &AssetBrowserFilterModel::filterChanged, this, &AssetBrowserTableModel::beginResetModel);
-            connect(m_filterModel, &AssetBrowserFilterModel::filterChanged, this, &AssetBrowserTableModel::UpdateTableModelMaps);
+            // This lambda starts a timer that protects against rapidly successive calls to UpdateTableModelMaps
+            auto checkUpdateModelTimer = [&]()
```

Comment:
suggestion/nit: It can be helpful to be explicit about which members to capture (this potentially could be reduced to `this` (not a huge deal, I just always prefer the explicitness of naming the variables to capture)

Reply:
I too prefer explicitness in general. Looks like I'll have at least one more revision coming relating to unit tests. I'll get this changed as well!

Reply:
Awesome stuff, thank you! 🙂